src/containers/SignInScreen/index.js

```diff
@@ -23,14 +24,19 @@ class SignInScreen extends Component {
   }
 
   render(){
-    const { signIn } = this.props;
+    const { signIn, facebookSignIn, loading, authenticated } = this.props;
+
+    if (loading || authenticated) {
+      return <Loading />;
+    }
+
     return (
       <View style={styles.signInContainer}>
         <View style={styles.safeAreaTop}>
           <ImageBackground source={signInBackground} style={styles.container} resizeMode='cover'>
             <View style={styles.formContainer}>
               <Text style={styles.targetMvdText}>TARGET MVD</Text>
-              <SignInForm onSubmit={user => signIn(user.toJS())}  signUpLink={this.navigateToSignUp}/>
+              <SignInForm onSubmit={user => signIn(user.toJS())}  goToSignUp={this.navigateToSignUp} facebookSignIn={ accessToken => facebookSignIn(accessToken) }/>
```

Comment:
```suggestion
              <SignInForm onSubmit={user => signIn(user.toJS())}  goToSignUp={this.navigateToSignUp} facebookSignIn={ accessToken => facebookSignIn(accessToken) }/>
```
Break this line into multiple